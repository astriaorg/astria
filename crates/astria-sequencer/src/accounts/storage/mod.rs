pub(super) mod keys;
mod values;

pub(crate) use values::Value;
pub(super) use values::{
    Balance,
    Nonce,
};
