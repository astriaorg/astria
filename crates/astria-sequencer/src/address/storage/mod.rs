pub(super) mod keys;
mod values;

pub(super) use values::AddressPrefix;
pub(crate) use values::Value;
