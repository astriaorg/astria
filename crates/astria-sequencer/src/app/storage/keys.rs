pub(in crate::app) const CHAIN_ID: &str = "app/chain_id";
pub(in crate::app) const REVISION_NUMBER: &str = "app/revision_number";
pub(in crate::app) const BLOCK_HEIGHT: &str = "app/block_height";
pub(in crate::app) const BLOCK_TIMESTAMP: &str = "app/block_timestamp";
pub(in crate::app) const CONSENSUS_PARAMS: &str = "app/consensus_params";

pub(in crate::app) fn storage_version_by_height(height: u64) -> String {
    format!("app/storage_version/{height}")
}

#[cfg(test)]
mod tests {
    use super::*;

    const COMPONENT_PREFIX: &str = "app/";

    #[test]
    fn keys_should_not_change() {
        insta::assert_snapshot!("chain_id_key", CHAIN_ID);
        insta::assert_snapshot!("revision_number_key", REVISION_NUMBER);
        insta::assert_snapshot!("block_height_key", BLOCK_HEIGHT);
        insta::assert_snapshot!("block_timestamp_key", BLOCK_TIMESTAMP);
        insta::assert_snapshot!("consensus_params_key", CONSENSUS_PARAMS);
        insta::assert_snapshot!("storage_version_key", storage_version_by_height(42));
    }

    #[test]
    fn keys_should_have_component_prefix() {
        assert!(CHAIN_ID.starts_with(COMPONENT_PREFIX));
        assert!(REVISION_NUMBER.starts_with(COMPONENT_PREFIX));
        assert!(BLOCK_HEIGHT.starts_with(COMPONENT_PREFIX));
        assert!(BLOCK_TIMESTAMP.starts_with(COMPONENT_PREFIX));
        assert!(CONSENSUS_PARAMS.starts_with(COMPONENT_PREFIX));
        assert!(storage_version_by_height(42).starts_with(COMPONENT_PREFIX));
    }
}
