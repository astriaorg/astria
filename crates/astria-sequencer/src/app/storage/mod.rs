mod values;

pub(crate) use values::{
    BlockHeight,
    BlockTimestamp,
    ChainId,
    RevisionNumber,
    StorageVersion,
    Value,
};
