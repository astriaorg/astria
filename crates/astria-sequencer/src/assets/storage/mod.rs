pub(super) mod keys;
mod values;

pub(super) use values::TracePrefixedDenom;
pub(crate) use values::Value;
