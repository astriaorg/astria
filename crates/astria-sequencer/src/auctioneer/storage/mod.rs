pub(super) mod keys;
mod values;

pub(super) use values::EnshrinedAuctioneerEntry;
pub(crate) use values::Value;
