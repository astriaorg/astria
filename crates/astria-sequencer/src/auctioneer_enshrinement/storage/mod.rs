pub(super) mod keys;
mod values;