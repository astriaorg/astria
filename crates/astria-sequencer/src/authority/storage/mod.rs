mod values;

pub(crate) use values::{
    AddressBytes,
    ValidatorSet,
    Value,
};
