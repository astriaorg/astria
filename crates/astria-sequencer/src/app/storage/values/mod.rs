mod block_height;
mod block_timestamp;
mod chain_id;
mod revision_number;
mod storage_version;

use borsh::{
    BorshDeserialize,
    BorshSerialize,
};

pub(in crate::app) use self::{
    block_height::BlockHeight,
    block_timestamp::BlockTimestamp,
    chain_id::ChainId,
    revision_number::RevisionNumber,
    storage_version::StorageVersion,
};

#[derive(Debug, BorshSerialize, BorshDeserialize)]
pub(crate) struct Value<'a>(ValueImpl<'a>);

#[derive(Debug, BorshSerialize, BorshDeserialize)]
enum ValueImpl<'a> {
    ChainId(ChainId<'a>),
    RevisionNumber(RevisionNumber),
    BlockHeight(BlockHeight),
    BlockTimestamp(BlockTimestamp),
    StorageVersion(StorageVersion),
}

#[cfg(test)]
mod tests {
    use std::mem::discriminant;

    use insta::assert_snapshot;

    use super::*;

    #[test]
    fn value_impl_chain_id_discriminant_unchanged() {
        assert_snapshot!(
            "value_impl_chain_id_discriminant",
            format!(
                "{:?}",
                discriminant(&ValueImpl::ChainId(
                    (&tendermint::chain::Id::try_from("test_prefix".to_string()).unwrap()).into()
                ))
            )
        );
    }

    #[test]
    fn value_impl_revision_number_discriminant_unchanged() {
        assert_snapshot!(
            "value_impl_revision_number_discriminant",
            format!("{:?}", discriminant(&ValueImpl::RevisionNumber(1.into())))
        );
    }

    #[test]
    fn value_impl_block_height_discriminant_unchanged() {
        assert_snapshot!(
            "value_impl_block_height_discriminant",
            format!("{:?}", discriminant(&ValueImpl::BlockHeight(1.into())))
        );
    }

    #[test]
    fn value_impl_block_timestamp_discriminant_unchanged() {
        assert_snapshot!(
            "value_impl_block_timestamp_discriminant",
            format!(
                "{:?}",
                discriminant(&ValueImpl::BlockTimestamp(tendermint::Time::now().into()))
            )
        );
    }

    #[test]
    fn value_impl_storage_version_discriminant_unchanged() {
        assert_snapshot!(
            "value_impl_storage_version_discriminant",
            format!("{:?}", discriminant(&ValueImpl::StorageVersion(1.into())))
        );
    }

    #[test]
    fn stored_value_app_discriminant_unchanged() {
        use crate::storage::StoredValue;
        assert_snapshot!(
            "stored_value_app_discriminant",
            format!(
                "{:?}",
                discriminant(&StoredValue::App(Value(ValueImpl::BlockHeight(1.into()))))
            )
        );
    }
}
