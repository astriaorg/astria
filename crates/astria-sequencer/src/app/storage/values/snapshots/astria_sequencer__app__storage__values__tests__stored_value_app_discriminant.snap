---
source: crates/astria-sequencer/src/app/storage/values/mod.rs
expression: "format!(\"{:?}\",\ndiscriminant(&StoredValue::App(Value(ValueImpl::BlockHeight(1.into())))))"
---
Discriminant(8)
